src/thumbnailer.cc

```diff
@@ -277,16 +290,19 @@ Thumbnailer::Status Thumbnailer::GenerateAnimationEqualQuality(
     }
   }
 
-  int curr_ind = 0;
-  for (auto& frame : frames_) {
-    frame.final_quality = final_quality;
-    frame.config.quality = final_quality;
-    CHECK_THUMBNAILER_STATUS(
-        GetPictureStats(curr_ind++, &frame.encoded_size, &frame.final_psnr));
+  for (std::size_t i = 0; i < frames_.size(); ++i) {
+    if (!frames_[i].near_lossless && frames_[i].final_quality < final_quality) {
```

Comment:
Should `frames_[i].final_quality > final_quality` be tested also?

Reply:
I think (here)[https://github.com/googleinterns/thumbnailer/blob/0d899e58a3b9256da63b97dbcdecc8df4d4b49f7/src/thumbnailer.cc#L276] , there's no change if `frames_[i].final_quality > final_quality`, so it does not need to be tested.